grafana/config.libsonnet

```
@@ -0,0 +1,56 @@
+local k = import 'k.libsonnet';
+{
+  _config+:: {
+    grafana+: {
+      rootUrl: error 'Root URL required',
+      provisioningDir: '/etc/grafana/provisioning',
+      dashboardConfigMapCount: 1,
+      labels+: {
+        dashboards: [],
+        datasources: [],
+        notificationChannels: [],
+      },
+      grafana_ini+: {
+        sections+: {
+          server: {
+            http_port: 3000,
+            router_logging: true,
+            root_url: $._config.grafana.rootUrl,
+          },
+          analytics: {
+            reporting_enabled: false,
+          },
+          users: {
+            default_theme: 'light',
+          },
+          explore+: {
+            enabled: true,
+          },
+        },
+      },
+    },
+  },
+
+  withGrafanaIniConfig(config):: {
+    _config+:: {
+      grafana+: {
```

```suggestion
      grafana_ini+: {
```
The two keys `grafana` and `grafana_ini` are confusing and error prone.

---

as described above, they both have valid meaning - I'm open to better names, these are the best I managed to come up with so far.